pkg/interactive/validation.go

```diff
@@ -63,14 +64,21 @@ func IsURL(val interface{}) error {
 }
 
 func IsValidHostname(val interface{}) error {
-	if !_isValidHostname(val.(string)) {
+	hostname := val.(string)
+	if hostname == "github.com" || strings.HasSuffix(hostname, ".github.com") {
+		return fmt.Errorf(fmt.Sprintf("'%s' hostname cannot be equal [*.]github.com", hostname))
+	}
+	if !_isValidHostname(hostname) {
 		return fmt.Errorf(fmt.Sprintf("'%s' hostname must be a valid DNS subdomain or IP address", val.(string)))
 	}
 	return nil
 }
 
 // _isValidHostname is same validation as in the Open Shift GitHub IDP CRD
+// Validates the hostname DNS1123 format
 // https://github.com/openshift/kubernetes/blob/91607f5d750ba4002f87d34a12ae1cfd45b45b81/openshift-kube-apiserver/admission/customresourcevalidation/oauth/helpers.go#L13
+// and denies the [*.]github.com hostnames
+// https://github.com/openshift/kubernetes/blob/258f1d5fb6491ba65fd8201c827e179432430627/openshift-kube-apiserver/admission/customresourcevalidation/oauth/validate_github.go#L49
 //
 //nolint:lll
 func _isValidHostname(hostname string) bool {
```

Comment:
@mnecas Personally I'd just collapse `_isValidHostname` into `IsValidHostname` unless there's a specific reason to keep it as a standalone  function?

Reply:
no specific reason I was just following the _isUrl but agree